src/compiler/scala/tools/nsc/backend/jvm/BTypesFromSymbols.scala

```diff
@@ -131,7 +131,8 @@ abstract class BTypesFromSymbols[G <: Global](val global: G) extends BTypes {
 
   def bootstrapMethodArg(t: Constant, pos: Position): AnyRef = t match {
     case Constant(mt: Type) => methodBTypeFromMethodType(transformedType(mt), isConstructor = false).toASMType
```

Comment:
While you are at it, this could be tweaked to also support [`Class` constants](https://docs.oracle.com/javase/specs/jvms/se16/html/jvms-4.html#jvms-4.4.1)
```suggestion
    case Constant(t: Type) =>
      transformedType(t) match {
        case mt: MethodType => methodBTypeFromMethodType(mt, isConstructor = false).toASMType
        case t: Type => typeToBType(t).toASMType
      }

```